content/en/agent/guide/secrets-management.md

```diff
@@ -216,79 +216,118 @@ instances:
 
-##### Usage
+##### Docker Swarm secrets
+[Docker swarm secrets][4] are mounted in the `/run/secrets` folder. So the Docker secret `db_prod_passsword` would be located in `/run/secrets/db_prod_password` of the Agent container. This would be referenced in the configuration by `ENC[file@/run/secrets/db_prod_password]`.
 
-The script requires a folder passed as an argument. Secret handles are interpreted as file names, relative to this folder. The script refuses to access any file out of the root folder (including symbolic link targets) to avoid leaking sensitive information.
+#### Read from Kubernetes Secret example
+This setup allows the Agent to directly read Kubernetes Secrets within its own and *other* namespaces. However, to do so the Agent's `ServiceAccount` must be granted permissions with the appropriate `Roles` and `RoleBindings` to do so.
 
-This script is incompatible with [OpenShift restricted SCC operations][3] and requires that the Agent runs as the `root` user.
+If the `Secret: database-secret` existed in the `Namespace: database`, and contained the data `password: example` this would be referenced in the configuration by `ENC[k8s_secret@database/database-secret/password]`. This will have the Agent pull this Secret directly from Kubernetes, which can be helpful when referencing a Secret that exists in a different namespace than the Agent is in.
 
-Starting with version 6.10.0, `ENC[]` tokens in config values passed as environment variables are supported. Previous versions only support `ENC[]` tokens found in `datadog.yaml` and in Autodiscovery templates.
+This does require additional permissions that are manually granted to the Agent's Service Account. For example the RBAC policy of:
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: datadog-secret-reader
+  namespace: database
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    resourceNames: ["database-secret"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: datadog-read-secrets
+  namespace: example
+subjects:
+  - kind: ServiceAccount
+    name: datadog-agent
+    apiGroup: ""
+    namespace: default
+roleRef:
+  kind: Role
+  name: datadog-secret-reader
+  apiGroup: ""
+```
+This `Role` would give access to the `Secret: database-secret` in the `Namespace: database`. The `RoleBinding` links up this permission to the `ServiceAccount: datadog-agent` in the `Namespace: default`. This will need to be manually added to your cluster with respect to your resources deployed.
+
+### (Legacy) Scripts for reading from files
+In version 7.32 of the Agent the script `readsecret_multiple_providers.sh` was added. This is the recommended script to use, rather than the legacy ones introduced in 6.12,  `/readsecret.py` and `/readsecret.sh`. However, `/readsecret.py` and `/readsecret.sh` are still included and supported in the Agent to read files.
```

Comment:
```suggestion
Datadog Agent v7.32 introduces the `readsecret_multiple_providers.sh` script. Datadog recommends that you use this script instead of `/readsecret.py` and `/readsecret.sh` from Agent v6.12. Note that `/readsecret.py` and `/readsecret.sh` are still included and supported in the Agent to read files.
```